aloe/result.py

```diff
@@ -0,0 +1,216 @@
+        if feature.tags:
+            term.writeln(term.cyan(feature.represent_tags()))
+
+        lines = feature.represented(annotate=False).splitlines()
+        term.writeln(term.bold_white(lines[0]))
+        term.writeln()
+        term.writeln(term.white('\n'.join(lines[1:])))
+        term.writeln()
+
+        yield
+    finally:
+        pass
+
+
+@outer_around.each_example
+@contextmanager
+@OutputTerminal.required
+def example_wrapper(term, scenario, outline, steps):
+    """Display scenario execution."""
+
+    try:
+        if scenario.tags:
+            term.writeln(term.cyan(scenario.represent_tags()))
+
+        start, end = scenario.represented().rsplit('#')
+        term.write(term.bold_white(start))
+        term.writeln(term.color(8)(end))
+
+        if outline:
+            term.writeln(represent_table([outline.keys(),
+                                          outline.values()],
+                                         indent=6,
+                                         cell_wrap=term.white))
+            term.writeln()
+
+        # write a preview of the steps
+        if term.is_a_tty:
+            steps_ = []
+
+            if scenario.feature.background:
+                steps_ += scenario.feature.background.steps
+
+            steps_ += steps
+
+            term.writeln(
+                term.color(8)('\n'.join(
+                    step.represented(annotate=False)
+                    for step in steps_
+                ) + '\n'),
+                return_=True
+            )
+
+        yield
+    finally:
+        term.writeln()
+
+
+@outer_around.each_step
+@contextmanager
+@OutputTerminal.required
+@hook_not_reentrant  # don't display inner steps called by top-level steps
+def step_wrapper(term, step):
+    """Display step execution."""
+
+    try:
+        if term.is_a_tty:
+            term.writeln(
+                step.represented(annotate=False, color=term.color(11)),
+                return_=True)
+
+        yield
+    finally:
+        if step.passed:
+            color = term.bold_green
+        elif step.failed:
+            color = term.bold_red
+        else:
+            color = term.yellow
+
+        term.writeln(
+            step.represented(annotate=False, color=color)
+        )
+
+
+class AloeTestResult(TextTestResult):
+    """
+    Cucumber test progress display (verbosity level 3).
+    """
+    # pylint:disable=too-many-arguments
+    def __init__(self, stream, descriptions, verbosity,
+                 config=None, errorClasses=None):
+        super().__init__(stream, descriptions, verbosity,
+                         config=config, errorClasses=errorClasses)
+        self.showAll = verbosity == 2
+        self.showSteps = verbosity >= 3  # pylint:disable=invalid-name
+
+        if self.showSteps:
+            TERMINAL[0] = OutputTerminal(
+                stream=stream,
+                force_styling=config.force_color)
+        else:
+            TERMINAL[0] = None  # unset the global
```

Comment:
Please don't comment on _what_ you are doing - that much is obvious from the code. _Why_ is this global passed around? (Because you can't register the steps early enough, etc.)